src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -1309,17 +1309,35 @@ GenTree* Lowering::LowerHWIntrinsic(GenTreeHWIntrinsic* node)
                 unsigned    simdSize        = node->GetSimdSize();
                 var_types   simdType        = Compiler::getSIMDTypeForSize(simdSize);
                 GenTree*    trueMask        = comp->gtNewSimdAllTrueMaskNode(simdBaseJitType, simdSize);
-                GenTree*    trueVal         = node;
-                GenTree*    falseVal        = comp->gtNewZeroConNode(simdType);
 
-                GenTreeHWIntrinsic* condSelNode =
-                    comp->gtNewSimdHWIntrinsicNode(simdType, trueMask, trueVal, falseVal, NI_Sve_ConditionalSelect,
-                                                   simdBaseJitType, simdSize);
 
-                BlockRange().InsertBefore(node, trueMask);
-                BlockRange().InsertBefore(node, falseVal);
-                BlockRange().InsertAfter(node, condSelNode);
-                use.ReplaceWith(condSelNode);
+                if (HWIntrinsicInfo::IsEmbeddedMaskForScalarResultOperation(intrinsicId))
+                {
+                    // Create the same node with an additional operand to pass the mask.
```

Comment:
can you add more details here calling it why we need extra operand for mask? Something along the lines: The instruction uses "predicate" to pick lanes, but at the same time returns scalar result. As such, we cannot wrap it inside `ConditionalSelect()` , because that node operates on `TYP_SIMD`. Hence, we will just add an operand so that we have a predicate register for such instructions.
`ExtractLastVector(TrueAll, value)`

Reply:
Sure, makes sense.